src/test/javascript/spec/component/exam/manage/student-exams/user-import-dialog.component.spec.ts

```diff
@@ -174,36 +179,35 @@ describe('UsersImportButtonComponent', () => {
         const studentsNotFound: StudentDTO[] = [{ registrationNumber: '3', firstName: 'Some', lastName: 'Dude', login: 'login3' }];
 
         const fakeResponse = { body: studentsNotFound } as HttpResponse<StudentDTO[]>;
-        sinon.replace(examManagementService, 'addStudentsToExam', sinon.fake.returns(of(fakeResponse)));
+        jest.spyOn(examManagementService, 'addStudentsToExam').mockReturnValue(of(fakeResponse));
 
         component.usersToImport = studentsToImport;
 
         fixture.detectChanges();
 
-        expect(component.hasImported).to.be.false;
-        expect(component.isSubmitDisabled).to.be.false;
+        expect(component.hasImported).toBeFalse;
+        expect(component.isSubmitDisabled).toBeFalse;
         const importButton = fixture.debugElement.query(By.css('#import'));
-        expect(importButton).to.exist;
+
+        expect(importButton).not.toBeNull;
 
         importButton.nativeElement.click();
 
-        expect(examManagementService.addStudentsToExam).to.have.been.calledOnce;
-        expect(component.isImporting).to.be.false;
-        expect(component.hasImported).to.be.true;
-        expect(component.notFoundUsers.length).to.equal(studentsNotFound.length);
+        expect(examManagementService.addStudentsToExam).toHaveBeenCalled();
+        expect(component.isImporting).toBe(false);
+        expect(component.hasImported).toBe(true);
+        expect(component.notFoundUsers.length).toEqual(studentsNotFound.length);
 
-        // Reset call counter
-        sinon.restore();
-        sinon.replace(examManagementService, 'addStudentsToExam', sinon.fake.returns(of(fakeResponse)));
+        jest.spyOn(examManagementService, 'addStudentsToExam').mockReturnValue(of(fakeResponse));
 
         component.hasImported = true;
         fixture.detectChanges();
 
         const finishButton = fixture.debugElement.query(By.css('#finish-button'));
-        expect(finishButton).to.exist;
+        expect(finishButton).not.toBeNull;
 
         finishButton.nativeElement.click();
 
-        expect(examManagementService.addStudentsToExam).to.not.have.been.called;
+        expect(examManagementService.addStudentsToExam).toHaveBeenCalled();
```

Comment:
Can you specify the number of calls here?

Reply:
Done